### PR TITLE
Update nemolite2d PSyKAl transformation to OpenCL

### PR DESCRIPTION
**Pull request by sergisiso**
Changes:
- Update tool-chain to Xilinx Vitis (new software stack)
- Add 'KernelGlobalsToArguments' transformation to the opencl psyclone script
- Remove the now unneeded 'rdt' from the continuity arguments.

Note that the 3rd change is necessary because an algorithm layer 'rdt' argument and promoting an 'rdt' import to the psy-layer can not coexist without renaming one of them. There is no proper renaming in Psyclone yet and looking at the manual_version calls to the continuity kernels seems that the argument was not there in the first place.

### CODE REVIEW

**Comment by sergisiso** [comment]
@arporter Can you check this small PR that applies the KernelGlobalsToArguments to the OpenCL script?

File: ocean/nemo/nemolite2d/psykal/Makefile [hunk 1]
```diff
@@ -111,12 +111,12 @@ nemolite2d_acc: common_lib timer_lib serial_inf_lib nemolite2d_alg.f90 acc_trans
 nemolite2d_ocl: common_lib timer_lib ocl_inf_lib nemolite2d_alg.f90 ocl_transform.py
 	@echo "Generating NemoLite2D with OpenCL"; rm -rf $@; mkdir -p $@
 	${PSYCLONE} -nodm -s ./ocl_transform.py --kernel-renaming single \
+		-I ${COMMON_DIR} \
 	   	-okern $@ -oalg $@/alg.f90 -opsy $@/psy.f90 nemolite2d_alg.f90
 	cp ${INF_LIB} $@/. # Each generated folder needs its purposely built inf_lib
 	cp namelist $@/.
 	cp Makefile_gen $@/Makefile
 	# Currently we need all kernels and the global parameters in a single OpenCL file
-	cat kernel_parameters.h > $@/allkernels
```
Comment (arporter) [inline ocean/nemo/nemolite2d/psykal/Makefile]: Can we remove `rdt` from `kernel_parameters.h` now?

File: ocean/nemo/nemolite2d/psykal/Makefile [hunk 2]
```diff
@@ -111,12 +111,12 @@ nemolite2d_acc: common_lib timer_lib serial_inf_lib nemolite2d_alg.f90 acc_trans
 nemolite2d_ocl: common_lib timer_lib ocl_inf_lib nemolite2d_alg.f90 ocl_transform.py
```
Comment (arporter) [inline ocean/nemo/nemolite2d/psykal/Makefile]: Please extend comment to say we use `-I` to tell PSyclone where to find the module file `USE`d by a kernel.

File: ocean/nemo/nemolite2d/psykal/ocl_transform.py [hunk 3]
```diff
@@ -5,24 +5,17 @@ def trans(psy):
...
-    # First invoke
+    # Get the invoke
     schedule = psy.invokes.get('invoke_0').schedule
-    cltrans.apply(schedule)
-
-    # Map of OpenCL options
-    ocl_options =  {'local_size': 4}
 
-    # Attach options object to Kernel Call and Kernel Schedule
+    # Remove the globals from inside each kernel
```
Comment (arporter) [inline ocean/nemo/nemolite2d/psykal/ocl_transform.py]: Is there a reason for removing the setting of 'local_size'?

File: ocean/nemo/nemolite2d/kernels/fortran/continuity_mod.f90 [hunk 4]
```diff
@@ -180,10 +178,10 @@ end subroutine invoke_continuity_arrays_basic
 
   subroutine continuity_code(ji, jj,                     &
                              ssha, sshn, sshn_u, sshn_v, &
-                             hu, hv, un, vn, rdt, e12t)
```
Comment (arporter) [inline ocean/nemo/nemolite2d/kernels/fortran/continuity_mod.f90]: Please could you ensure that the various places that call this kernel (in e.g. the manual versions) are updated to remove the rdt argument. e.g. psykal_acc/time_step_mod.f90

File: ocean/nemo/nemolite2d/psykal/Makefile_gen [hunk 5]
```diff
@@ -61,13 +61,13 @@ nemolite2dxilinx: ${KERNELS} allkernels.xclbin psy.o alg.o
...
 %.o: %.mod
 
-# Xilinx OpenCL kernel compilation
+# Xilinx Vitis OpenCL kernel compilation
 %.xclbin: %.cl
 	# Compile to a .xo file
-	xocc --compile --platform ${PLATFORM} --target ${EXECUTION_TARGET} \
+	v++ --compile --platform ${PLATFORM} --target ${EXECUTION_TARGET} \
 		$< -o $<.xo
 	# Link to crete a Xilinx FPGA bianry container (.xclbin)
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply ocean/nemo/nemolite2d/psykal/Makefile_gen]: Fixed

File: ocean/nemo/nemolite2d/psykal/Makefile (hunk 1 above)
Comment (sergisiso) [reply ocean/nemo/nemolite2d/psykal/Makefile]: In fact we can remove the whole `kernel_parameters.h` now. I forgot about it.

File: ocean/nemo/nemolite2d/psykal/ocl_transform.py (hunk 3 above)
Comment (sergisiso) [reply ocean/nemo/nemolite2d/psykal/ocl_transform.py]: I couldn't see any performance improvement from it, so for the baseline version I prefer to go with the defaults. I plan to write an optimized psyclone script in the future, but first I need to find which parameters/values provide a real benefit.

File: ocean/nemo/nemolite2d/kernels/fortran/continuity_mod.f90 (hunk 4 above)
Comment (sergisiso) [reply ocean/nemo/nemolite2d/kernels/fortran/continuity_mod.f90]: True, I updated some. But note that the one you pointed on `psykal_acc/time_step_mod.f90` has its own redefined `continuity_code`.

File: ocean/nemo/nemolite2d/psykal/Makefile (hunk 2 above)
Comment (sergisiso) [reply ocean/nemo/nemolite2d/psykal/Makefile]: Done

**Comment by sergisiso** [comment]
@arporter Ready for next review

File: ocean/nemo/nemolite2d/psykal/ocl_transform.py (hunk 3 above)
Comment (arporter) [reply ocean/nemo/nemolite2d/psykal/ocl_transform.py]: Makes sense.

File: ocean/nemo/nemolite2d/psykal/Makefile (hunk 1 above)
Comment (arporter) [reply ocean/nemo/nemolite2d/psykal/Makefile]: Great.

File: ocean/nemo/nemolite2d/kernels/fortran/continuity_mod.f90 (hunk 4 above)
Comment (arporter) [reply ocean/nemo/nemolite2d/kernels/fortran/continuity_mod.f90]: Thanks.

**Pull request merged**